docs/developer/index.rst

```diff
@@ -736,6 +761,7 @@ with:
 A regularly-gridded 2D dataset can be created, but note that the
 arguments must be flattened:
 
+  >>> import numpy as np
```

Comment:
Here's questions I have about the testing of each "snippet":

- when do symbols have to be imported or created?
  - Imports: I assume that we don't need to do it for each test, so that this import will cover follow-up tests, but if I just had a section at the top of the page (as I do in some places) that says something like "the following examples assumes these imports ,... and then imports a bunch of modules", then we can use them in all following tests
  - symbols: if I define "foo" somewhere as a variable, is it then present for later tests?

Reply:
rst files are run like one big single piece of code form top to bottom. You have access to everything (imports and variables) that was defined before. There is special markup to define things in a way that doctest knows about is, but sphinx does not show it and also the possibility or a prefix (e.g. run "import numpy as np" before every file), but I have not used either of them here: https://github.com/astropy/pytest-doctestplus